test/WarmResumptionTest.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include <thread>
+
+#include <gtest/gtest.h>
+
+#include <folly/SocketAddress.h>
+
+#include "rsocket/RSocket.h"
+#include "rsocket/RSocketServiceHandler.h"
+#include "rsocket/transports/tcp/TcpConnectionAcceptor.h"
+#include "rsocket/transports/tcp/TcpConnectionFactory.h"
+#include "yarpl/flowable/TestSubscriber.h"
+
+using namespace rsocket;
+using namespace yarpl::flowable;
+
+namespace {
+class HelloStreamRequestResponder : public RSocketResponder {
+ public:
+  yarpl::Reference<Flowable<rsocket::Payload>> handleRequestStream(
+      rsocket::Payload request,
+      rsocket::StreamId) override {
+    auto requestString = request.moveDataToString();
+    return Flowables::range(1, 10)->map([name = std::move(requestString)](
+        int64_t v) {
+      std::stringstream ss;
+      ss << "Hello " << name << " " << v << "!";
+      std::string s = ss.str();
+      return Payload(s, "metadata");
+    });
+  }
+};
+
+class HelloServiceHandler : public RSocketServiceHandler {
+ public:
+  folly::Expected<RSocketConnectionParams, RSocketException> onNewSetup(
+      const SetupParameters&) override {
+    return RSocketConnectionParams(
+        std::make_shared<HelloStreamRequestResponder>());
+  }
+
+  void onNewRSocketState(
+      std::shared_ptr<RSocketServerState> state,
+      ResumeIdentificationToken token) override {
+    store_.lock()->insert({token, std::move(state)});
+  }
+
+  folly::Expected<std::shared_ptr<RSocketServerState>, RSocketException>
+  onResume(ResumeIdentificationToken token) override {
+    auto itr = store_->find(token);
+    CHECK(itr != store_->end());
+    return itr->second;
+  };
+
+ private:
+  folly::Synchronized<
+      std::map<ResumeIdentificationToken, std::shared_ptr<RSocketServerState>>,
+      std::mutex>
+      store_;
+};
+
+std::shared_ptr<RSocketClient> makeResumableClient(uint16_t port) {
```

Comment:
Do you think this should be in `test/RSocketTests.h`instead?  I could see it being used from more than just this test file.